cvmfs/test/repository_test.py

```diff
@@ -95,3 +95,34 @@ def test_expired_whitelist(self):
         self.assertRaises(cvmfs.RepositoryVerificationFailed,
                           cvmfs.open_repository,
                           self.mock_repo.dir, public_key=self.mock_repo.public_key)
+
+
+    def test_lookup(self):
+        self.mock_repo.make_valid_whitelist()
+        self.mock_repo.serve_via_http()
+        repo = cvmfs.open_repository(self.mock_repo.url,
+                                     public_key=self.mock_repo.public_key)
+        dirent = repo.lookup('/.cvmfsdirtab')
+        self.assertIsNotNone(dirent)
+        dirent = repo.lookup('/bar/4/foo')
+        self.assertIsNotNone(dirent)
+        dirent = repo.lookup('/bar/4/foobar')
+        self.assertIsNone(dirent)
+
+
+    def test_list(self):
+        self.mock_repo.make_valid_whitelist()
+        self.mock_repo.serve_via_http()
+        repo = cvmfs.open_repository(self.mock_repo.url,
+                                     public_key=self.mock_repo.public_key)
+        dirents = repo.list_directory('/')
+        self.assertIsNotNone(dirents)
+        self.assertEqual(3, len(dirents))
+        dirents = repo.list_directory('/bar/3')
+        self.assertIsNotNone(dirents)
+        self.assertEqual(4, len(dirents))
+        dirents = repo.list_directory('/bar/4/foo')
+        self.assertIsNone(dirents)
+        dirents = repo.list_directory('/fakedir')
+        self.assertIsNone(dirents)
```

Comment:
Yes, they do. Only one level, though (because there are no more levels at the moment)